### PR TITLE
*: Add the arguments of `ctx` and `etcdCli` to the `NewDDL` interface

### PR DESCRIPTION
**Pull request by zimulala**
Add the arguments of `ctx` and `etcdCli` to the interface.
Remove the function of `setWork`.

### CODE REVIEW

**Comment by hanfei1991** [comment]
I think you should store the context in ddl structure. Because in `close` function, you should cancel by the context. In this way, you needn't pass another context to `SetLease` function.

**Comment by shenli** [comment]
Why should we add ctx in NewDDL interface? What's is that used for?

**Comment by zimulala** [comment]
Go documentation is described as follows: 
`Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. `@hanfei1991 

Many etcd interfaces require this argument. @shenli

**Comment by hanfei1991** [comment]
@zimulala But do we really need a context to pass ? In other words, need we close the ddl structure outside?

**Comment by hanfei1991** [comment]
Currently, we have both quitCh and context. Which one is in charge of sending the close signal ?

**Comment by zimulala** [comment]
@hanfei1991 
We need the context as an argument to etcd interface.
`context` only use to pass information to etcd and inform the `worker` to exit.  Its use is not the same as `quitCh`.

File: ddl/ddl.go [hunk 1]
```diff
@@ -295,8 +293,12 @@ func (d *ddl) Stop() error {
 	return errors.Trace(err)
 }
 
-func (d *ddl) start() {
+func (d *ddl) start(ctx goctx.Context) {
 	d.quitCh = make(chan struct{})
+	if ChangeOwnerInNewWay {
+		d.campaignOwners(ctx)
```
Comment (shenli) [inline ddl/ddl.go]: Can we use goctx.Background() here?

**Comment by shenli** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL @tiancaiamao @hanfei1991

**Pull request merged**